galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.cemt.manager/src/main/java/dev/galasa/cicsts/cemt/ICEMT.java

```diff
@@ -13,31 +15,99 @@
 
 public interface ICEMT {
 
+
+   /** 
+    * Inquire a CEMT resource using the resource type and name.
+    * This does not support inquiries of multiple resources at once. 
+    * Will return {@link null} if the resource is not found.
+    * @param cemtTerminal an {@link ITerminal} object logged on to the CICS region and in an active CEMT session.
+    * If mixed case is required, the terminal should be presented with no upper case translate status. 
+    * For example, the test could first issue <code>CEOT TRANIDONLY</code>
+    * @param resourceType a {@link String} of the resource type you are looking for.
+    * @param resourceName a {@link String} of the name of the resource you are looking for.
+    * @return a {@link HashMap} object containing all of the properties of the resource.
+    * @throws CEMTException if resource not found.
+    */
+
    
+
    public HashMap<String, String> inquireResource(@NotNull ITerminal cemtTerminal,
                                                   @NotNull String resourceType,
                                                   @NotNull String resourceName
                                                   ) throws CEMTException;
    
-   
+
+   /** 
+    * Set the state of a CEMT resource using the resource type and name. If the searchText is not found on the terminal screen then
+    * an exception will be thrown.
+    * @param cemtTerminal an {@link ITerminal} object logged on to the CICS region and in an active CEMT session.
+    * If mixed case is required, the terminal should be presented with no upper case translate status. 
+    * For example, the test could first issue <code>CEOT TRANIDONLY</code>
+    * @param resourceType a {@link String} of the type of resource you want to set.
+    * @param resourceName a {@link String} of the name of the resource you want to set.
+    * @param action a {@link String} of the action you want to perform.
+    * @param searchText a {@link String} of the text you want to search for on the terminal screen.
+    * @throws CEMTException
+    */
+
    public void setResource(@NotNull ITerminal cemtTerminal,
                            @NotNull String resourceType,
                                     String resourceName,
```

Comment:
dont you need notnull on this as well?

Reply:
No because resourceName can be null, as you can CEMT SET ALL